code/game/objects/items/devices/scanners.dm

```diff
@@ -931,9 +931,23 @@ GENE SCANNER
 		return
 	buffer = C.dna.mutation_index
 	to_chat(user, "<span class='notice'>Subject [C.name]'s DNA sequence has been saved to buffer.</span>")
-	if(LAZYLEN(buffer))
-		for(var/A in buffer)
-			to_chat(user, "<span class='notice'>[get_display_name(A)]</span>")
+
+	var/list/full_list_mutations = list()
+	for(var/each in buffer) // get inherent mutations first
+		full_list_mutations[each] = FALSE
+	for(var/datum/mutation/each_mutation in C.dna.mutations)
+		if(each_mutation.type in buffer) // active inherent mutation
+			full_list_mutations[each_mutation.type] = "Activated"
+		else // active artificial mutation
+			full_list_mutations[each_mutation.type] = "Injected"
+	for(var/datum/mutation/each_mutation in C.dna.temporary_mutations)
+		full_list_mutations[each_mutation.type] = "Temporary"
+
+	for(var/A in full_list_mutations)
```

Comment:
This will still show all mutations on a person, it just won't show if they are permanent or temporary, not sure if you wanted to remove showing active mutations as a result of the previous comment. If not I don't see a reason not to show if its active or not

Reply:
> it just won't show if they are permanent or temporary
Temporary isn't a functional code, but I implemented it anyway.

> not sure if you wanted to remove showing active mutations as a result of the previous comment. If not I don't see a reason not to show if its active or not
Tsar-Salat said the thing I made here is closer to balance change, so I removed a feature where a scanner can see if it's activated or injected.
meanwhile, it can tell more active genes than before - current code doesn't tell injected genes, but the PR code will include the injected genes (but will not say if it's injected)